subsys/fs/littlefs_fs.c

```diff
@@ -801,7 +805,9 @@ struct fs_mount_t FS_FSTAB_ENTRY(DT_DRV_INST(inst)) = { \
 	.type = FS_LITTLEFS, \
 	.mnt_point = DT_INST_PROP(inst, mount_point), \
 	.fs_data = &fs_data_##inst, \
-	.storage_dev = (void *)DT_FIXED_PARTITION_ID(FS_PARTITION(inst)), \
+	.storage_dev = (void *) (USE_PARTITION_MANAGER ?\
```

Comment:
This looks nice!

Is it possible to use `( defined(USE_PARTITION_MANAGER) ? \` instead of having the `#define USE_PARTITION_MANAGER 0` above?

Reply:
I am not sure whether `defined` is evaluated when not used with `#if`. I am quite sure that up to c99 the `defined` would be left there after pre-processor phase.

Reply:
Think `IS_ENABLED(USE_PARTITION_MANAGE)` will does the job

Reply:
Maybe `IF_ENABLED` ?

Reply:
Seems that there is need for `DEFINED()` macro :D